flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/ComputationState.java

```diff
@@ -114,6 +115,28 @@ public DeweyNumber getVersion() {
 		return version;
 	}
 
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof ComputationState) {
+			ComputationState other = (ComputationState) obj;
+			return Objects.equals(state, other.state) &&
+				event == other.event &&
```

Comment:
The combination of `Object.equals(event, other.event)` and `count == other.counter` tells exactly that. I would not base the implementation on reference equity.